packages/astro/src/core/dev/update-check.ts

```diff
@@ -0,0 +1,44 @@
+import { fetchPackageJson } from '../../cli/install-package.js';
+import type { AstroPreferences } from '../../preferences/index.js';
+
+export const MAX_PATCH_DISTANCE = 5; // If the patch distance is less than this, don't bother the user
+const CHECK_MS_INTERVAL = 1_036_800_000; // 12 days, give or take
+
+let _latestVersion: string | undefined = undefined;
+
+export async function fetchLatestAstroVersion(
+	preferences: AstroPreferences | undefined
+): Promise<string> {
+	if (_latestVersion) {
+		return _latestVersion;
+	}
+
+	const packageJson = await fetchPackageJson(undefined, 'astro', 'latest');
+	if (packageJson instanceof Error) {
+		throw packageJson;
+	}
+
+	const version = packageJson?.version;
+
+	if (!version) {
+		throw new Error('Failed to fetch latest Astro version');
+	}
+
+	if (preferences) {
+		await preferences.set('checkUpdates._lastCheck', Date.now(), { reloadServer: false });
```

Comment:
So, I had a choice here:
- I add one more file
- I re-use one of the files we have

To me, it didn't feel really wrong to use preferences for this, since users are not supposed to change it by hand. So I went with putting it in preferences, that way I don't have to essentially redo preferences again, but let me know if I should add a third store somewhere for this kind of data, happy to do either (and can be changed any time anyway, not like it's breaking)